lib/api/controllers/adminController.js

```diff
@@ -98,57 +106,84 @@ class AdminController {
         result.deletedRoles = deletedRoles;
         return this.kuzzle.internalEngine.bootstrap.createDefaultProfiles();
       })
-      .then(() => this.kuzzle.internalEngine.bootstrap.createDefaultRoles());
-
-    if (refresh === 'wait_for') {
-      return promise.then(() => result);
-    }
+      .then(() => this.kuzzle.internalEngine.bootstrap.createDefaultRoles())
+      .then(() => delete _locks.resetSecurity)
+      .catch(error => {
+        delete _locks.resetSecurity;
+        throw error;
+      });
 
-    return Bluebird.resolve({ acknowledge: true });
+    return this._waitForAction(request, promise, result);
   }
 
   /**
    * Reset all indexes created by users
    */
-  resetDatabase () {
+  resetDatabase (request) {
+    this._lockAction(request, 'Kuzzle is already reseting all indexes.');
+
     const indexes = Object.keys(this.kuzzle.indexCache.indexes).filter(idx => idx[0] !== '%');
 
-    Bluebird.map(indexes, index => {
-      const request = new Request({ index });
+    const promise = Bluebird.map(indexes, index => {
+      const req = new Request({ index, refresh: request.input.args.refresh });
```

Comment:
Shouldn't we rather always force a refresh on all resetXX actions?
I cannot see any use case where a user would drop some content and would still like to get some possible search results within the next second.

Reply:
Actually, the user would expect that the data would no more be available straight away. So the wait_for should probably be the default bahavior